CMakeLists.txt

```
@@ -131,9 +141,16 @@ target_link_directories(DirectX5::DirectX5 INTERFACE "${CMAKE_SOURCE_DIR}/3rdpar
 add_library(Vec::Vec INTERFACE IMPORTED)
 target_include_directories(Vec::Vec INTERFACE "${CMAKE_SOURCE_DIR}/3rdparty/vec")
 
-add_library(lego1 SHARED
-  LEGO1/main.cpp
-)
+if (EMSCRIPTEN)
+  add_library(lego1 STATIC
+    LEGO1/main.cpp
+  )
+else()
+  add_library(lego1 SHARED
+    LEGO1/main.cpp
+  )
+endif()
+
```

Depending on `BUILD_SHARED_LIBS`, this will create a shared or static library.
```suggestion
add_library(lego1
  LEGO1/main.cpp
)

```